hydrocron/api/controllers/authorizer.py

```diff
@@ -0,0 +1,66 @@
+"""
+Lambda Authorizer to facilitate usage of API keys and usage plans.
+
+Taken from example:
+https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-use-lambda-authorizer.html
+"""
+
+import json
+import logging
+
+from hydrocron.utils import connection
+
+
+logging.getLogger().setLevel(logging.INFO)
+
+
+def authorization_handler(event, context):
+    """Lambda authorizer function to allow or deny a request."""
+
+    logging.info("Event: %s", event)
+    logging.info("Context: %s", context)
+
+    api_key_trusted = "" if "x-hydrocron-key" not in event["headers"].keys() else event["headers"]["x-hydrocron-key"]
+    stored_api_key_trusted = get_api_key("trusted")
+
+    if api_key_trusted and api_key_trusted == stored_api_key_trusted:
+        response_policy = create_policy("trusted_partner", "Allow", event["methodArn"], stored_api_key_trusted)
+        logging.info("Created policy for truster partner.")
+
+    else:
+        stored_api_key_default = get_api_key("default")
+        response_policy = create_policy("default_user", "Allow", event["methodArn"], stored_api_key_default)
+        logging.info("Created policy for default user.")
+
+    logging.info("Response: %s", response_policy)
+    return json.loads(response_policy)
+
+
+def get_api_key(key_type):
```

Comment:
If we are able to do this lookup during lambda initialization it will help reduce the number of calls to SSM. Since we do not plan to change the api key frequently, we shouldn't have to worry about getting out of sync

Reply:
Moved the API key look up to the Lambda initialization and confirmed the correct API key was being pulled and used against the correct usage plan.